src/Controllers/ImageController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace JustBetter\GlideDirective\Controllers;
+
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+use League\Glide\Server;
+use League\Glide\Signatures\Signature;
+use League\Glide\Signatures\SignatureException;
+use Statamic\Assets\Asset;
+use Statamic\Contracts\Assets\Asset as AssetContract;
+use Statamic\Facades\Asset as AssetFacade;
+use Statamic\Imaging\ImageGenerator;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+
+class ImageController extends Controller
+{
+    protected ?AssetContract $asset;
+
+    protected array $params;
+
+    public function __construct(protected ImageGenerator $imageGenerator, protected Server $server) {}
+
+    public function getImageByPreset(Request $request, string $preset, string $fit, string $signature, string $file, string $format): BinaryFileResponse
+    {
+        /** @var ?Asset $asset */
+        $asset = AssetFacade::findByUrl(Str::start($file, '/'));
+        $this->asset = $asset;
+        $this->params = ['s' => $signature, 'preset' => $preset, 'fit' => $fit, 'format' => $format];
```

Comment:
Maybe after merge; We can create a ImageRequest to validate the parameters, now it will probably have an empty string in one of these variables if not filled.

Reply:
If one of these parameters is not filled it means the given path is not right, which already results in a 404.
Haven't been able to get this to go wrong at least.